core/src/main/scala/org/apache/spark/executor/CoarseGrainedExecutorBackend.scala

```diff
@@ -41,7 +41,8 @@ private[spark] class CoarseGrainedExecutorBackend(
     executorId: String,
     cores: Int,
     userClassPath: Seq[URL],
-    env: SparkEnv)
+    env: SparkEnv,
+    hostname: String)
```

Comment:
@XuTingjun Before this patch, the hostname was set by CoarseGrainedSchedulerBackend. But now, it just uses the specified one. It would crease some problem. E.g., if the specified `hostname` is `0.0.0.0`, `Utils.localHostName/localHostNameForURI` may not work correctly as it doesn't return a real host or ip.